ledger/participant-integration-api/src/main/scala/platform/store/cache/MutableCacheBackedContractStore.scala

```diff
@@ -102,11 +99,12 @@ private[platform] class MutableCacheBackedContractStore(
             // Since this contract is part of the input, it was able to be looked up once.
             // So this is the case of a divulged contract, which was not archived.
             // Divulged contract does not change maximumLedgerTime
+
+            // TODO LLP: Consider making this stack-safe
```

Comment:
Consider the extreme case of an exercise that archives 10_000 contracts just after a Ledger API start when the caches are cold. This would recurse 10000 times no?

Reply:
Hmm, I initially thought this is on-stack, but the recursion is actually inside a Future(..).flatMap so it is safe. Sorry for the noise, will remove